packages/typespec-azure-resource-manager/lib/common-types/types.tsp

```diff
@@ -11,24 +12,34 @@ model ArmResourceBase {}
  * Base model that defines common properties for all Azure Resource Manager resources.
  */
 @doc("Common properties for all Azure Resource Manager resources.")
-model ArmResource extends ArmResourceBase {
+model Resource extends ArmResourceBase {
   @doc("Fully qualified resource ID for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}")
   @visibility("read")
-  id: string;
+  id?: armResourceIdentifier;
```

Comment:
Hmm, yeah, I agree that we should describe this correctly, but I wonder if it won't cause some breaking changes for some sdks based on the v3 definitions

Reply:
You are right, This was changed to `arm-id` in v4. While it doesn't impact current specs, I will use versioning to correctly model it. Fixing.